### PR TITLE
Add Gen AI toggle between edit/presentation mode

### PR DESCRIPTION
**Pull request by fisher-alice**
This PR adds a toggle between 'Edit' mode and 'User View'/presentation mode. When in presentation mode, the Instructions and Model Customization panels are replaced by the Model Card panel. I access the `hidePresentationPanel` level property and if this value is set to `true`, the toggle is no longer displayed and only 'Edit' mode is available. I set this value to `true` for 's/allthethings/lessons/47/levels/2'.

I also access the `botName` level property to set the name of the chat workspace in presentation mode.

Screencast video AFTER updates:

https://github.com/code-dot-org/code-dot-org/assets/107423305/6c1ebfb2-ee7c-44e3-a8af-84df8c14e20c

## Links
[jira](https://codedotorg.atlassian.net/browse/LABS-550)
Note - I am breaking up this jira ticket into parts.
Next task is to create the `PresentationView` component using collapsible sections.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/aichat/views/AichatView.tsx [hunk 1]
```diff
@@ -41,34 +51,80 @@ const AichatView: React.FunctionComponent = () => {
...
+  const aiCustomizations = useAppSelector(
+    state => state.aichat.currentAiCustomizations
+  );
+
+  const viewModeButtonsProps: SegmentedButtonsProps = {
+    buttons: [
+      {
+        label: 'Edit',
+        value: ViewMode.EDIT,
+        iconLeft: {iconName: 'wrench', iconStyle: 'solid', title: 'check'},
```
Comment (bencodeorg) [inline apps/src/aichat/views/AichatView.tsx]: Is "check" the right title here? Reading about semantic icons here:

https://docs.fontawesome.com/web/dig-deeper/accessibility#icons-used-as-semantic-elements

File: apps/src/aichat/views/AichatView.tsx [hunk 2]
```diff
@@ -41,34 +51,80 @@ const AichatView: React.FunctionComponent = () => {
...
+      },
+      {
+        label: 'User View',
+        value: ViewMode.PRESENTATION,
+        iconLeft: {iconName: 'user-group', iconStyle: 'solid', title: 'check'},
+      },
+    ],
+    size: 'm',
+    selectedButtonValue: viewMode,
+    onChange: (value: string) => setViewMode(value),
```
Comment (bencodeorg) [inline apps/src/aichat/views/AichatView.tsx]: Nit: just `onChange: setViewMode`?

**Review by bencodeorg — APPROVED** [review]
LGTM!

File: apps/src/aichat/views/AichatView.tsx (hunk 1 above)
Comment (fisher-alice) [reply apps/src/aichat/views/AichatView.tsx]: And thanks for the reference - very useful!

**Pull request merged**